datafusion/proto/proto/datafusion.proto

```diff
@@ -369,7 +369,8 @@ message LogicalExprNode {
 }
 
 message Wildcard {
-  string qualifier = 1;
+  string qualifier = 1 [deprecated = true];
+  TableReference relation = 2;
```

Comment:
Theoretically this should be a `oneof`, but I'm not sure if adding `oneof` will alter the pbjson serialization. Since we are deprecating the field anyway, I decided to handle the disambiguation in `from_proto.rs` and `to_proto.rs` instead.

Also, I'm not sure if we are allowed to make breaking changes to protobuf here. If we do, I can simply change the type of `qualifier` to make the code simpler.

Reply:
> Also, I'm not sure if we are allowed to make breaking changes to protobuf here. If we do, I can simply change the type of qualifier to make the code simpler.

At the moment, we don't try to maintain any backwards compatibility, thus I think we should opt for simpler code

https://github.com/apache/datafusion/blob/8aad2082994bdea14eaf7d0e8c177aa3ce9af67b/datafusion/proto/src/lib.rs#L37-L42